### DISCUSSION TITLE
add_middleware causes exceptions to not be raised/handled silently

### DISCUSSION THREAD

**Discussion by fraser-langton** [fastapi/fastapi#6256] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import uvicorn
from fastapi import FastAPI
from starlette.middleware.base import BaseHTTPMiddleware


app = FastAPI()


@app.get("/info")
def info():
    # raises Exception as expected, the traceback is seen in console
    raise Exception


private_api = FastAPI()


@private_api.get("/info")
def info():
    # exception is handled silently, no traceback is seen in console
    raise Exception


app.mount("/private", private_api)


class Middleware(BaseHTTPMiddleware):

    async def dispatch(self, request, call_next):
        return await call_next(request)


app.add_middleware(Middleware)  # when this is removed, the exceptions are raised for all routes


if __name__ == '__main__':
    uvicorn.run(app, port=8000)
```

### Description

When exceptions are raised on a subapp, the exceptions are not propagated all the way to see in console
In the example code 
**/info** raises an exception and the full stacktrace is seen in console
**/private/info** does not raise the exception and only `INFO:     127.0.0.1:56308 - "GET /info HTTP/1.1" 500 Internal Server Error` is shown in console

### Operating System

Windows

### FastAPI Version

0.74.0

### Python Version

3.8.10

### Additional Context

uvicorn==0.17.6

**Comment by fraser-langton** [comment]
Upgrading to 0.79.0 the issue goes away, not sure if this was fixed intentionally (I couldn't find a closed issue) or inadvertently in which case might still be an issue elsewhere

**Comment by iudeen** [comment]
What's the behavior if you add the Middleware to both of your apps? 🤔

**Comment by Kludex** [comment, marked as answer]
> Upgrading to 0.79.0 the issue goes away, not sure if this was fixed intentionally (I couldn't find a closed issue) or inadvertently in which case might still be an issue elsewhere

It was solved on Starlette.

**Comment by fraser-langton** [comment]
> It was solved on Starlette.
Thanks @Kludex - was very confused!

**Comment by fraser-langton** [comment]
Issue is back see my versions below, should I open an issue on starlette? 

fastapi==0.88.0
starlette==0.23.0       
uvicorn==0.20.0

**Comment by Kludex** [comment]
You can open a discussion on starlette.

**Comment by fraser-langton** [comment]
Opened discussion in [starlette ](https://github.com/encode/starlette/discussions/1976)
